napi/parser/wrap.cjs

```diff
@@ -25,39 +23,32 @@ module.exports.wrap = function wrap(result) {
   };
 };
 
+// Set `value` field of `Literal`s which are `BigInt`s or `RegExp`s.
+//
+// Returned JSON contains an array `fixes` with paths to these nodes
+// e.g. for `123n; foo(/xyz/)`, `fixes` will be
+// `[["body", 0, "expression"], ["body", 1, "expression", "arguments", 2]]`.
+//
+// Walk down the AST to these nodes and alter them.
+// Compiling the list of fixes on Rust side avoids having to do a full AST traversal on JS side
+// to locate the likely very few `Literal`s which need fixing.
 function jsonParseAst(programJson) {
-  const program = JSON.parse(programJson);
-  transformNode(program);
+  const { node: program, fixes } = JSON.parse(programJson);
+  for (const fixPath of fixes) {
+    applyFix(program, fixPath);
+  }
   return program;
```

Comment:
I have no idea as to the perf implications of this... but, couldn't we just do this in the reviver of `JSON.parse`

effectivly
```
function jsonParseAst(programJson) {
   const { node: program, fixes } = JSON.parse(programJson, (key, value) => {
// apply fix in here
});
   return program;
```

this way you only visit the tree once?

(note: i guess it might depend on the runtime's implementation of JSON.parse - the overhead of the native to JS call for the reviver might actually make this slower

Reply:
Hahaha! Until earlier today that's exactly what we had. This whole business was prompted by the discovery that using a `JSON.parse` reviver was producing a 3x slow-down. See #10783.

Reply:
aha I see!